src/amplitude.js

```diff
@@ -99,10 +70,11 @@ Amplitude.prototype.init = function(apiKey, opt_userId, opt_config, callback) {
     _upgradeCookeData(this);
     _loadCookieData(this);
 
-    this.options.deviceId = (opt_config && opt_config.deviceId !== undefined &&
-        opt_config.deviceId !== null && opt_config.deviceId) ||
-        this.options.deviceId || UUID();
-    this.options.userId = (opt_userId !== undefined && opt_userId !== null && opt_userId) || this.options.userId || null;
+    // load deviceId and userId from input, or try to fetch existing value from cookie
+    this.options.deviceId = (type(opt_config) === 'object' && type(opt_config.deviceId) === 'string' &&
+        !utils.isEmptyString(opt_config.deviceId) && opt_config.deviceId) || this.options.deviceId || UUID() + 'R';
```

Comment:
Appending a 'R' to generated deviceIds to be consistent with iOS and Android SDKs